### PR TITLE
Move entropy plugin to main aspect

### PR DESCRIPTION
**Pull request by RanpengLi**
I move the files from the entropy plugin in the benchmark folder to the main aspect

### For all pull requests:

* [X] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

### CODE REVIEW

**Review by jdannberg — APPROVED** [review]
Looks good to me!

**Comment by RanpengLi** [comment]
@tjhei Hi Timo, my changes are failing on the Mac tester but we don't know why. Do you have any idea on why it is happening and how could I fix it? Thank you!

Ranpeng

**Comment by tjhei** [comment]
This is the error:
```
Undefined symbols for architecture arm64:
  "aspect::MaterialModel::PlasticAdditionalOutputs<2>::PlasticAdditionalOutputs(unsigned int)", referenced from:
      aspect::MaterialModel::EntropyModel<2>::create_additional_named_outputs(aspect::MaterialModel::MaterialModelOutputs<2>&) const in unity_12_cxx.cxx.o
  "aspect::MaterialModel::PlasticAdditionalOutputs<3>::PlasticAdditionalOutputs(unsigned int)", referenced from:
      aspect::MaterialModel::EntropyModel<3>::create_additional_named_outputs(aspect::MaterialModel::MaterialModelOutputs<3>&) const in unity_12_cxx.cxx.o
ld: symbol(s) not found for architecture arm64
```

I assume you are missing some explicit instantiations.

**Comment by RanpengLi** [comment]
We just tried adding the following lines to the visco_plastic.cc for the explicit instantiations, but it failed with the same error message. Do you have any hints on what else is missing?

```
#define INSTANTIATE(dim) \
  template class PlasticAdditionalOutputs<dim>;

    ASPECT_INSTANTIATE(INSTANTIATE)

#undef INSTANTIATE   
```

File: source/material_model/visco_plastic.cc [hunk 1]
```diff
@@ -685,5 +685,12 @@ namespace aspect
                                    "The value for the components of this formula and additional "
                                    "parameters are read from the parameter file in subsection "
                                    " 'Material model/Visco Plastic'.")
+
+#define INSTANTIATE(dim) \
+  template class PlasticAdditionalOutputs<dim>;
+
+    ASPECT_INSTANTIATE(INSTANTIATE)
+
+#undef INSTANTIATE
```
Comment (gassmoeller) [inline source/material_model/visco_plastic.cc:694]: I think I know what happened. This code needs to go into the same source file that defines `PlasticAdditionalOutputs`. That is `source/material_model/rheology/visco_plastic.cc`, not this one. Please check that, and if true, move the code into the other file.

Comment (RanpengLi) [reply source/material_model/visco_plastic.cc]: It works!! Thank you!

**Review by gassmoeller — APPROVED** [review]

**Pull request merged**